controllers/utility.go

```diff
@@ -0,0 +1,52 @@
+package controllers
+
+import (
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/giantswarm/apiextensions/v3/pkg/annotation"
+	"github.com/giantswarm/apiextensions/v3/pkg/label"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func defaultRequeue() reconcile.Result {
+	return ctrl.Result{
+		Requeue:      true,
+		RequeueAfter: time.Minute * 5,
+	}
+}
+
+func timedRequeue(upgradeTime time.Time) reconcile.Result {
+	if time.Until(upgradeTime) > 5 {
+		return defaultRequeue()
+	}
+	return ctrl.Result{
+		Requeue:      true,
+		RequeueAfter: time.Until(upgradeTime),
```

Comment:
I wonder if we maybe wanna add like +1 sec to ensure that it `time.After()` will be always true when we execute this requeue

or do you think there is no need and we can rely on that the timing will be exactly right?

Reply:
You have a point there. Let's just add a second to be sure.